core/src/main/java/org/elasticsearch/snapshots/SnapshotsService.java

```diff
@@ -578,8 +581,28 @@ private SnapshotInfo inProgressSnapshot(SnapshotsInProgress.Entry entry) {
                         shardSnapshotStatus.failure(shardFailure.reason());
                         shardStatus.put(shardId, shardSnapshotStatus);
                     } else {
-                        IndexShardSnapshotStatus shardSnapshotStatus =
-                            repository.getShardSnapshotStatus(snapshotInfo.snapshotId(), snapshotInfo.version(), indexId, shardId);
+                        IndexShardSnapshotStatus shardSnapshotStatus;
+                        if (snapshotInfo.state() == SnapshotState.FAILED) {
+                            // If the snapshot failed, but the shard's snapshot does
+                            // not have an exception, it means that partial snapshots
+                            // were disabled and in this case, the shard snapshot will
+                            // *not* have any metadata, so attempting to read the shard
+                            // snapshot status will throw an exception.  Instead, we create
+                            // a status for the shard to indicate that the shard snapshot
+                            // could not be taken due to partial being set to false.
+                            shardSnapshotStatus = new IndexShardSnapshotStatus();
+                            shardSnapshotStatus.updateStage(IndexShardSnapshotStatus.Stage.FAILURE);
+                            String msg = "snapshot not taken because partial snapshots are not " +
+                                         "configured for this snapshot and another shard caused " +
+                                         "the snapshot to fail";
```

Comment:
good idea, i think we can just keep this as "skipped" for now, as that captures the essence of what happened with that shard.